text/src/od.rs

```diff
@@ -0,0 +1,718 @@
+                    .to_string(),
+            );
+        }
+
+        if self.octal_bytes {
+            self.type_strings = vec!["o1".to_string()];
+        }
+        if self.unsigned_decimal_words {
+            self.type_strings = vec!["u2".to_string()];
+        }
+        if self.octal_words {
+            self.type_strings = vec!["o2".to_string()];
+        }
+        if self.signed_decimal_words {
+            self.type_strings = vec!["d2".to_string()];
+        }
+        if self.hex_words {
+            self.type_strings = vec!["x2".to_string()];
+        }
+
+        // Check if multiple mutually exclusive options are used together
+        let mut basic_types = 0;
+        if self.octal_bytes {
+            basic_types += 1;
+        }
+        if self.bytes_char {
+            basic_types += 1;
+        }
+        if self.unsigned_decimal_words {
+            basic_types += 1;
+        }
+        if self.octal_words {
+            basic_types += 1;
+        }
+        if self.signed_decimal_words {
+            basic_types += 1;
+        }
+        if self.hex_words {
+            basic_types += 1;
+        }
+
+        if basic_types > 1 {
+            return Err(
+                "Options '-b', '-c', '-d', '-o', '-s', '-x' cannot be used together".to_string(),
+            );
+        }
+
+        Ok(())
+    }
+}
+
+/// Parses an offset string and converts it into a `u64` value.
+///
+/// # Parameters
+///
+/// - `offset: &str`: A string slice representing the offset. This string can be in hexadecimal
+///   format prefixed with "0x" or "0X", octal format prefixed with "0", or decimal format. The
+///   string may also end with 'b', 'k', or 'm' to indicate byte multipliers.
+///
+/// # Returns
+///
+/// - `Result<u64, Box<dyn std::error::Error>>`: This function returns a `Result` which is:
+///   - `Ok(u64)`: On success, the parsed and multiplied offset as a `u64`.
+///   - `Err(Box<dyn std::error::Error>)`: On failure, an error boxed as a `dyn std::error::Error`.
+///
+fn parse_skip(offset: &str) -> Result<u64, Box<dyn std::error::Error>> {
+    let (number, multiplier) = if offset.starts_with("0x") || offset.starts_with("0X") {
+        // For hexadecimal, 'b' should be part of the number if it is the last character
+        (offset, 1)
+    } else if offset.ends_with('b') {
+        (&offset[..offset.len() - 1], 512)
+    } else if offset.ends_with('k') {
+        (&offset[..offset.len() - 1], 1024)
+    } else if offset.ends_with('m') {
+        (&offset[..offset.len() - 1], 1048576)
+    } else {
+        (offset, 1)
+    };
+
+    let base_value = if number.starts_with("0x") || number.starts_with("0X") {
+        u64::from_str_radix(&number[2..], 16)?
+    } else if number.starts_with('0') && number.len() > 1 {
+        u64::from_str_radix(&number[1..], 8)?
+    } else {
+        number.parse()?
+    };
+
+    Ok(base_value * multiplier)
```

Comment:
Very similar to `parse_count` fn. Possibly use generics to specify to which type we should parse (usize/u64/etc)